pkg/common/config/config.go

```diff
@@ -477,12 +480,14 @@ func GetSupervisorNamespace(ctx context.Context) (string, error) {
 }
 
 // GetClusterFlavor returns the cluster flavor based on the env variable set in the driver deployment file
-func GetClusterFlavor(ctx context.Context) cnstypes.CnsClusterFlavor {
+func GetClusterFlavor(ctx context.Context) (cnstypes.CnsClusterFlavor, error) {
 	// CLUSTER_FLAVOR is defined only in Supervisor and Guest cluster deployments.
 	// If it is empty, it is implied that cluster flavor is Vanilla K8S
 	clusterFlavor := cnstypes.CnsClusterFlavor(os.Getenv("CLUSTER_FLAVOR"))
 	if strings.TrimSpace(string(clusterFlavor)) == "" {
```

Comment:
if someone specify vanilla as the CLUSTER_FLAVOR, it will fail. if none is specified or vanilla is specified means vanilla. and if other than guest and workloadmit is Unrecognized.

Reply:
For Vanilla, we are not publishing any yaml file with the CLUSTER_FLAVOR env. Are you considering a case where users go ahead and add a new env in the yaml ?

Reply:
yes but still we should not fail if someone specifies this.

we have this defined in the govmomi - https://github.com/vmware/govmomi/blob/master/cns/types/enum.go#L39

```
const (
	CnsClusterFlavorVanilla  = CnsClusterFlavor("VANILLA")
	CnsClusterFlavorWorkload = CnsClusterFlavor("WORKLOAD")
	CnsClusterFlavorGuest    = CnsClusterFlavor("GUEST_CLUSTER")
	CnsClusterFlavorUnknown  = CnsClusterFlavor("ClusterFlavor_Unknown")
)
```

Reply:
Done